src/GafferScene/Shader.cpp

```diff
@@ -275,6 +279,39 @@ class Shader::NetworkBuilder
 			}
 		}
 
+		IECoreScene::ShaderNetwork::Parameter parameterForPlug( const Plug *parameter )
```

Comment:
I wonder if `parameterForPlug()` implies a generality that this function doesn't actually have? All the places we call it, we've already verified that the parameter is an output, and the function body doesn't contain any logic for dealing with input parameters (which _are_ representable as `ShaderNetwork::Parameter`). So perhaps it should be called something like `outputParameterForPlug()`, and contain some assertions?